modules/creating-the-training-job.adoc

```diff
@@ -0,0 +1,163 @@
+
+[role='_abstract']
+Before you can use a training job to tune a model, you must create the training job. 
+The example training job in this section is based on the IBM and Hugging Face tuning example provided link:https://github.com/foundation-model-stack/fms-hf-tuning/tree/main/examples/prompt_tuning_twitter_complaints[here]. 
+
+
+.Prerequisites
+ifdef::upstream,self-managed[]
+* You have logged in to {openshift-platform} with the `cluster-admin` role.
+endif::[]
+ifdef::cloud-service[]
+* You have logged in to OpenShift with the `cluster-admin` role.
+endif::[]
+
+ifndef::upstream[]
+* You have access to a data science cluster that is configured to run distributed workloads as described in link:{rhoaidocshome}{default-format-url}/working_with_distributed_workloads/configuring-distributed-workloads_distributed-workloads[Configuring distributed workloads].
+endif::[]
+ifdef::upstream[]
+* You have access to a data science cluster that is configured to run distributed workloads as described in link:{odhdocshome}/working-with-distributed-workloads/#configuring-distributed-workloads_distributed-workloads[Configuring distributed workloads].
+endif::[]
+
+ifndef::upstream[]
+* You have created a data science project. 
+For information about how to create a project, see link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-on-data-science-projects_nb-server#creating-a-data-science-project_nb-server[Creating a data science project].
+endif::[]
+ifdef::upstream[]
+* You have created a data science project. 
+For information about how to create a project, see link:{odhdocshome}/working-on-data-science-projects/#_using_data_science_projects[Creating a data science project].
+endif::[]
+
+* You have Admin access for the data science project.
+** If you created the project, you automatically have Admin access. 
+** If you did not create the project, your cluster administrator must give you Admin access.
+
+* You have access to a model.
+* You have access to data to train the model.
+
+.Procedure
+. In a terminal window, if you are not already logged in to your OpenShift cluster as a cluster administrator, log in to the OpenShift CLI as shown in the following example:
++
+[source,subs="+quotes"]
+----
+$ oc login __<openshift_cluster_url>__ -u __<admin_username>__ -p __<password>__
+----
+
+. Define a `ConfigMap` object in a YAML file called `config_trainingjob.yaml` with the following contents:
++
+[source]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: my-config-trainingjob
+  namespace: kfto
+data:
+  config.json: |
+    {
+      "model_name_or_path": "bigscience/bloom-560m",
+      "training_data_path": "/data/input/twitter_complaints.json",
+      "output_dir": "/data/output/tuning/bloom-twitter",
+      "num_train_epochs": 10.0,
+      "per_device_train_batch_size": 4,
+      "gradient_accumulation_steps": 4,
+      "learning_rate": 1e-05,
+      "response_template": "\n### Label:",
```

Comment:
- **model_name_or_path**: Specifies the pre-trained model to be used (`bigscience/bloom-560m`).
- **training_data_path**: Path to the training data set previously in our data config map (tranining_data.yaml)
- **output_dir**: Our output directory for the model
- **num_train_epochs**: Specifies the number of epochs for training
- **per_device_train_batch_size**: Batch size (4) (How many example for the data set are processed together.
- **gradient_accumulation_steps**: Number of gradient accumulation steps (4).
- **learning_rate**: Learning rate for the training (1e-05).
- **response_template**: Template formatting for the response (`\n### Label:`).
- **dataset_text_field**: Field in the dataset to be used for training (`output`)(This can be seen in training_data.yaml).
- **use_flash_attn**: Flag for using flash attention.

Reply:
Thank you, updated in latest commit.